src/tools/util.rs

```diff
@@ -0,0 +1,29 @@
+/// Generate a random data.
+pub fn gen_rand_bytes(len: usize) -> Vec<u8> {
+    (0..len).map(|_| rand::random::<u8>()).collect()
```

Comment:
Will `fill` work? Like:
`let mut v = vec![0; 1024];
 thread_rng().fill(&mut v);`

Reply:
yeah, probably. but it requires a variable allocation...
I feel the current approach is more idiomatic